Code/GraphMol/Descriptors/BagOfBonds.h

```diff
@@ -0,0 +1,45 @@
+//
+//  Copyright (c) 2018, Guillaume GODIN
+//  All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are
+// met:
+//
+//     * Redistributions of source code must retain the above copyright
+//       notice, this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above
+//       copyright notice, this list of conditions and the following
+//       disclaimer in the documentation and/or other materials provided
+//       with the distribution.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+// "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+// LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+// A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+// OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+// LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+// DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+
+#ifndef BagOfBondsRDKIT_H_JUNE2018
+#define BagOfBondsRDKIT_H_JUNE2018
+
+#ifdef RDK_BUILD_DESCRIPTORS3D
+namespace RDKit {
+class ROMol;
+namespace Descriptors {
+const std::string BagOfBondsVersion = "1.0.0";
+
+	void BagOfBondsVector(const ROMol &mol, std::vector<double> &res, int confId,
+    int alpha, std::map<std::string, unsigned int> MaxBags);
+
+    std::map<std::string, unsigned int>  BagOfBondsMap(std::vector<std::string>  smiles);
```

Comment:
Why does this function need to be in the public API?
If it's going to be here, it should be documented.

Reply:
we need to get list of BoB vector before building the map

Reply:
ok